### DISCUSSION TITLE
About JackTrip license

### DISCUSSION THREAD

**Discussion by xnakos** [jacktrip/jacktrip#1196] [General]
Hi, JackTrippers!

I have a couple of questions on the licensing of JackTrip. It is mentioned that earlier versions (and files) were MIT, while later/contemporary files are under GPL. First of all, I would like to ask why the change occurred and, secondly, if there is an aggregate license for JackTrip, if that makes sense. The reason for my asking is understanding licensing effects and repercussions.

Thank you!

**Comment by mikedickey** [comment]
JackTrip's graphical user interfaces (basically everything inside src/gui) is GPL licensed. The core components and command line utility (no gui mode) are MIT.

Note that even JackTrip's core components use Qt, so any application that builds on top of it is subject to the Qt license as well. You can purchase a commercial license for Qt; otherwise, it is subject to GPL.

I do hope someday to break out all the core components into a cleaner library, which has no Qt dependency and therefore is truly free of copyleft, but other priorities...
